portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,168 +49,203 @@
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
-
+ 
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private int numberOfCommentsToDisplay = 0;
-  private List<String> tagsToSearch = new ArrayList<String>();
   private final static String MESSAGE_PARAMETER = "text-input";
   private final static String SENDER_PARAMETER = "sender";
   private final static String TAG_PARAMETER = "tags";
-  
-  // TODO: ADD TO FORM IN INDEX
-  private final static long PARENT_ID_PARAMETER = 0;
-  
   public final static String BLOG_ENTITY_KIND = "blogMessage";
-  
-  private void putBlogsInDatastore(String tag, String message, String nickname, List<BlogMessage> reply, long parentID) {
-    Entity blogMessageEntity = new Entity(BLOG_ENTITY_KIND);
-    blogMessageEntity.setProperty("nickname", nickname);
-    blogMessageEntity.setProperty("text", message);
-    blogMessageEntity.setProperty("time", System.currentTimeMillis());
-    blogMessageEntity.setProperty("tag", tag);
-    blogMessageEntity.setProperty("replies", reply);
-    blogMessageEntity.setProperty("parentID", parentID);
-    
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(blogMessageEntity);
-  }
-
-  private List<BlogMessage> getBlogsFromDatastore() {
-    List<BlogMessage> BlogMessages = new ArrayList<BlogMessage>();
-    Query query = new Query(BLOG_ENTITY_KIND).addSort("time", SortDirection.ASCENDING);
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
-
-    UserService userService = UserServiceFactory.getUserService();
-
-    for (Entity entity : results.asIterable()) {
-      long messageId = entity.getKey().getId();
-      long timestamp = (long) entity.getProperty("time");
-      String tag = (String) entity.getProperty("tag");
-      String comment = (String) entity.getProperty("text");
-      String nickname = (String) entity.getProperty("nickname");
-      String email = (String) userService.getCurrentUser().getEmail();
-      long parentID = (long) 0;
-      // TODO: UPDATE WHEN FRONTEND INPUT IS TAKEN 
-      ArrayList<BlogMessage> messageReplies = (ArrayList) entity.getProperty("replies");
-      BlogMessage message = new BlogMessage(
-            messageId, tag, comment, nickname, email, messageReplies, timestamp, parentID);
-      BlogMessages.add(message);
-    }
-    return BlogMessages;
-  }
-    
+ 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get BlogMessages from Datastore.
-    List<BlogMessage> BlogMessages = getBlogsFromDatastore();
-
-    // Separate posts from replies from BlogMessages. 
+    List<BlogMessage> blogMessages = LoadAllBlogsOrLast(true);
```

Comment:
Nit: usually you would put a nested comment for constant parameter to make it easier to read like:

LoadAllBlogsOrLast(/*all=*/true)